testsuite/smoke-test/src/rest_api.rs

```diff
@@ -19,3 +23,59 @@ impl PublicUsageTest for GetIndex {
         Ok(())
     }
 }
+
+pub struct BasicClient;
+
+impl Test for BasicClient {
+    fn name(&self) -> &'static str {
+        "api::basic-client"
+    }
+}
+
+impl PublicUsageTest for BasicClient {
+    fn run<'t>(&self, ctx: &mut PublicUsageContext<'t>) -> Result<()> {
+        let runtime = tokio::runtime::Builder::new_current_thread()
+            .enable_all()
+            .build()?;
+
+        runtime.block_on(async move {
+            let client = Client::new(reqwest::Url::parse(ctx.rest_api_url()).unwrap());
+
+            client.get_ledger_information().await.unwrap();
+
+            let mut account1 = ctx.random_account();
+            ctx.create_parent_vasp_account(account1.authentication_key())
+                .unwrap();
+            ctx.fund(account1.address(), 10).unwrap();
+            let account2 = ctx.random_account();
+            ctx.create_parent_vasp_account(account2.authentication_key())
+                .unwrap();
+            ctx.fund(account2.address(), 10).unwrap();
+
+            let tx =
+                account1.sign_with_transaction_builder(ctx.transaction_factory().peer_to_peer(
+                    diem_sdk::transaction_builder::Currency::XUS,
+                    account2.address(),
+                    1,
+                ));
+            client.submit(&tx).await.unwrap();
```

Comment:
here set pending_txn and use the hash to get txn.

get txn returns pending txn if it's in mempool, and returns user txn when it is executed, otherwise becomes not found error if txn is dropped by mempool

Reply:
Perfect, fixed this and included a simple wait_for_txn function that uses the rest api instead.